hydra_base/db/model/rule.py

```diff
@@ -31,10 +31,9 @@ class Rule(AuditMixin, Base, Inspect, PermissionControlled):
 
     __tablename__ = 'tRule'
     __table_args__ = (
-        UniqueConstraint('scenario_id', 'name', name="unique rule name"),
+        UniqueConstraint('network_id', 'name', name="unique rule name"),
```

Comment:
Would it make sense to have the same constraint for templates and projects? Seems arbitrary to have it for networks but not for other levels

Reply:
Yes, these should definitely be consistent: I've added this in 451e29e, along with a check to allow admin to read all rules, and a test for permissions.